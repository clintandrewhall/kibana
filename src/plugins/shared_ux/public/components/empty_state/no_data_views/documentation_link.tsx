/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import {
  EuiDescriptionList,
  EuiDescriptionListDescription,
  EuiDescriptionListTitle,
  EuiLink,
  useEuiTheme,
} from '@elastic/eui';
import React from 'react';

import { FormattedMessage } from '@kbn/i18n-react';
import { css } from '@emotion/css';

interface Props {
  documentationUrl: string;
}

export function DocumentationLink({ documentationUrl }: Props) {
  const { euiTheme } = useEuiTheme();
  const { colors, size, border } = euiTheme;
  const titleCSS = css`
    width: auto !important;
  `;
  const footerCSS = css`
    background: ${colors.lightestShade};
    margin: 0 (-${size.l}) (-${size.l});
    padding: ${size.l};
    border-radius: 0 0 ${border.radius.medium} ${border.radius.medium};
    align-items: baseline !important;
  `;
  return (
    <EuiDescriptionList className={footerCSS} type="responsiveColumn">
      <EuiDescriptionListTitle className={titleCSS}>
        <FormattedMessage
          id="sharedUX.noDataViews.learnMore"
          defaultMessage="Want to learn more?"
        />
      </EuiDescriptionListTitle>
      <EuiDescriptionListDescription>
        <EuiLink href={documentationUrl} target="_blank" external>
          <FormattedMessage
            id="sharedUX.noDataViews.readDocumentation"
            defaultMessage="Read documentation"
          />
        </EuiLink>
      </EuiDescriptionListDescription>
    </EuiDescriptionList>
  );
}
