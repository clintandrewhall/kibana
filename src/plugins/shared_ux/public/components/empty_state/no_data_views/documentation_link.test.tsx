/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import {
  EuiDescriptionList,
  EuiDescriptionListDescription,
  EuiDescriptionListTitle,
  EuiLink,
} from '@elastic/eui';
import { DocumentationLink } from './documentation_link';
import React from 'react';
import { mountWithIntl } from '@kbn/test-jest-helpers';

describe('<DocumentationLink />', () => {
  test('is rendered correctly', () => {
    const component = mountWithIntl(<DocumentationLink documentationUrl={'dummy'} />);
    expect(component).toMatchSnapshot();

    expect(component.find(EuiDescriptionList).length).toBe(1);
    expect(component.find(EuiDescriptionListTitle).length).toBe(1);
    expect(component.find(EuiDescriptionListDescription).length).toBe(1);

    const link = component.find(EuiLink).at(0);
    expect(link.prop('href')).toBe('dummy');
  });
});
