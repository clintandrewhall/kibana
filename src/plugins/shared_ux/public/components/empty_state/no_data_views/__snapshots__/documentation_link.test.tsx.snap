// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentationLink /> is rendered correctly 1`] = `
.emotion-0 {
  background: #f0f4fb;
  margin: 0 (-24px) (-24px);
  padding: 24px;
  border-radius: 0 0 6px 6px;
  -webkit-align-items: baseline!important;
  -webkit-box-align: baseline!important;
  -ms-flex-align: baseline!important;
  align-items: baseline!important;
}

.emotion-2 {
  width: auto!important;
}

<DocumentationLink
  documentationUrl="dummy"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
>
  <EuiDescriptionList
    className="emotion-0"
    type="responsiveColumn"
  >
    <dl
      className="euiDescriptionList euiDescriptionList--responsiveColumn emotion-0"
    >
      <EuiDescriptionListTitle
        className="emotion-2"
      >
        <dt
          className="euiDescriptionList__title emotion-2"
        >
          <FormattedMessage
            defaultMessage="Want to learn more?"
            id="sharedUX.noDataViews.learnMore"
            values={Object {}}
          >
            Want to learn more?
          </FormattedMessage>
        </dt>
      </EuiDescriptionListTitle>
      <EuiDescriptionListDescription>
        <dd
          className="euiDescriptionList__description"
        >
          <EuiLink
            external={true}
            href="dummy"
            target="_blank"
          >
            <a
              className="euiLink euiLink--primary"
              href="dummy"
              rel="noopener noreferrer"
              target="_blank"
            >
              <FormattedMessage
                defaultMessage="Read documentation"
                id="sharedUX.noDataViews.readDocumentation"
                values={Object {}}
              >
                Read documentation
              </FormattedMessage>
              <EuiIcon
                aria-label="External link"
                className="euiLink__externalIcon"
                size="s"
                type="popout"
              >
                <span
                  aria-label="External link"
                  className="euiLink__externalIcon"
                  data-euiicon-type="popout"
                  size="s"
                />
              </EuiIcon>
              <EuiScreenReaderOnly>
                <span
                  className="euiScreenReaderOnly"
                >
                  <EuiI18n
                    default="(opens in a new tab or window)"
                    token="euiLink.newTarget.screenReaderOnlyText"
                  >
                    (opens in a new tab or window)
                  </EuiI18n>
                </span>
              </EuiScreenReaderOnly>
            </a>
          </EuiLink>
        </dd>
      </EuiDescriptionListDescription>
    </dl>
  </EuiDescriptionList>
</DocumentationLink>
`;
